host-controller/src/main/java/org/jboss/as/host/controller/HostControllerService.java

```diff
@@ -128,6 +130,20 @@ public void start(StartContext context) throws StartException {
             b.append("Configured system properties:");
             for (String property : new TreeSet<String>(properties.stringPropertyNames())) {
                 String propVal = property.toLowerCase(Locale.getDefault()).contains("password") ? "<redacted>" : properties.getProperty(property, "<undefined>");
+                if (property.toLowerCase(Locale.getDefault()).equals("sun.java.command") && !propVal.isEmpty()) {
+                    Pattern pattern = Pattern.compile("(-D(?:[^ ])+=)((?:[^ ])+)");
+                    Matcher matcher = pattern.matcher(propVal);
+                    StringBuffer sb = new StringBuffer(propVal.length());
+                    while (matcher.find()) {
+                        String option = matcher.group(2);
+                        if (matcher.group(1).contains("password")) {
+                            option = "<redacted>";
+                        }
+                        matcher.appendReplacement(sb, Matcher.quoteReplacement(matcher.group(1) + option));
```

Comment:
I think we should just make this:
```
if (matcher.group(1).contains("password")) {
   matcher.appendReplacement(sb, Matcher.quoteReplacement(matcher.group(1) + "<redacted>"));
}
```
There's not a need to add the overhead of replacing all values.

Reply:
Corrected it. Thanks.